tests/test_reactions.py

```diff
@@ -0,0 +1,196 @@
+        )
+        self.helper.send_event(
+            room_id,
+            EventTypes.Reaction,
+            {
+                "m.relates_to": {
+                    "event_id": message_3_body.get("event_id"),
+                    # A single "compound" or grapheme cluster emoji is allowed
+                    # :male-technologist:
+                    # \uD83D\uDC68\u200D\uD83D\uDCBB
+                    # xf0x9fx91xa8xe2x80x8dxf0x9fx92xbb
+                    "key": "👨‍💻",
+                    "rel_type": RelationTypes.ANNOTATION,
+                }
+            },
+            tok=self.map_user_id_to_token[self.user_a],
+            expect_code=HTTPStatus.OK,
+        )
+
+    def test_empty_reaction(self) -> None:
+        room_id = self.create_local_room(self.user_a, [], False)
+        assert room_id is not None
+        message_1_body = self.helper.send(
+            room_id, "message 1", tok=self.map_user_id_to_token[self.user_a]
+        )
+        self.helper.send_event(
+            room_id,
+            EventTypes.Reaction,
+            {
+                "m.relates_to": {
+                    "event_id": message_1_body.get("event_id"),
+                    "key": "",
+                    "rel_type": RelationTypes.ANNOTATION,
+                }
+            },
+            tok=self.map_user_id_to_token[self.user_a],
+            # Need clarification on if this should pass or not
+            expect_code=HTTPStatus.OK,
+        )
+
+    def test_multiple_cluster_reaction(self) -> None:
+        room_id = self.create_local_room(self.user_a, [], False)
+        assert room_id is not None
+        message_1_body = self.helper.send(
+            room_id, "message 1", tok=self.map_user_id_to_token[self.user_a]
+        )
+        self.helper.send_event(
+            room_id,
+            EventTypes.Reaction,
+            {
+                "m.relates_to": {
+                    "event_id": message_1_body.get("event_id"),
+                    # Words(or sentences) are not allowed
+                    "key": "Hello there",
+                    "rel_type": RelationTypes.ANNOTATION,
+                }
+            },
+            tok=self.map_user_id_to_token[self.user_a],
+            expect_code=(
+                HTTPStatus.BAD_REQUEST
+                if self.limit_reactions_enabled
+                else HTTPStatus.OK
+            ),
+        )
+        message_2_body = self.helper.send(
+            room_id, "message 2", tok=self.map_user_id_to_token[self.user_a]
+        )
+        self.helper.send_event(
+            room_id,
+            EventTypes.Reaction,
+            {
+                "m.relates_to": {
+                    "event_id": message_2_body.get("event_id"),
+                    # Two separate emoji are not allowed(in the same reaction)
+                    # :+1: and :smiley:
+                    # \uD83D\uDC4D\uD83D\uDE03
+                    "key": "👍😃",
```

Comment:
can you test for a flag and a family emoji as well to validate, that those are allowed to be sent? (They are composed emojis)